engine/workflow_step_matrix_post_message.go

```diff
@@ -32,25 +32,12 @@ var getMatrixClient = func(homeserver string) (MatrixClient, error) {
 	return mc, nil
 }
 
-func (s postMessageMatrixWorkflowStep) getMatrixClient(e *engine) (mc MatrixClient, err error) {
+func (s postMessageMatrixWorkflowStep) getMatrixClient(e *engine) (mc matrix.Client, err error) {
 	if s.asBot != "" {
-		modelBot, err := e.botRepository.FindByIdentifier(s.asBot)
-		if err != nil {
-			return nil, err
-		}
-
-		// Convert model/bot to engine/bot
-		// TODO: Remove once engine/bot has been replaced in favour of model/bot
-		b := MakeBotFromModelBot(modelBot)
-
-		if !b.IsHydrated() {
-			b.Hydrate(e)
-		}
-
-		return b.getMCInstance(), nil
+		return s.botRegistry.GetPrimaryClient()
 	}
 
-	return e.client, nil
+	return s.botRegistry.GetClient(s.asBot)
```

Comment:
@psrpinto We did this wrong. Inside the `if` it should be a non primary bot and outside (default) it should be the primary client

Reply:
Great catch! Fixing.